### PR TITLE
Running EnginePlayer Async  | Fixed DustRevealer was working with normal glowstones

### PR DESCRIPTION
**Pull request by Mohamad82Bz**
Today I found that normal glowstones will trigger DustRevealer for everyone, I checked out the code and i saw the checks are wrong. There were a isDust method that was checking if the item was a glowstone and not a Dust Revealer custom item.

I also did some small changes, **UNTESTED** but I guess it's fine.
EDIT: I'm testing the changes in my public server, No issues so far.

In this pull request I just wanted to fix the isDust method, Sorry if I broke something

Thanks!

### CODE REVIEW

**Comment by Mohamad82Bz** [comment]
Description of my third commit:

A few days after my public server release, I noticed EnginePlayer is taking all of the server's performance, I checked it out and I saw It's okay to be ran Async, because it was all maths and calculations and just a spawnParticle, playSound method that weren't thread safe, I made everything Async except spawnParticle and playSound parts. I tested this change on my productive server and I didn't get any issues so far.

File: src/main/java/com/volmit/iris/object/IrisEffect.java [hunk 1]
```diff
@@ -228,7 +228,10 @@ public void apply(Player p,  GeneratorAccess g)
 		if(sound != null)
 		{
 			Location part = p.getLocation().clone().add(RNG.r.i(-soundDistance, soundDistance), RNG.r.i(-soundDistance, soundDistance), RNG.r.i(-soundDistance, soundDistance));
-			p.playSound(part, getSound(), (float) volume, (float) RNG.r.d(minPitch, maxPitch));
+
+			J.sr(() -> {
+				p.playSound(part, getSound(), (float) volume, (float) RNG.r.d(minPitch, maxPitch));
+			});
```
Comment (Mohamad82Bz) [inline src/main/java/com/volmit/iris/object/IrisEffect.java:239]: playSound part should be ran sync, So I used scheduler to run this statement sync in an Async method

Comment (cyberpwnn) [reply src/main/java/com/volmit/iris/object/IrisEffect.java:239]: I think `J.sr` is Sync Repeating. I think you want `J.s` there.

Comment (Mohamad82Bz) [reply src/main/java/com/volmit/iris/object/IrisEffect.java:239]: J.sr was a runTask method, I changed it as you said. Thanks

**Comment by cyberpwnn** [comment]
yeet

**Pull request merged**